Pattern from code bugfixes

emjin opened the pull request.

emjin committed:
Added string literals for the rest of languages

brendongo commented:
Is this for pattern generation? If so can you make that more clear in the PR title/body

emjin committed:
Implemented id_qualified in pretty_print_generic

emjin committed:
Support parsing constructors (IdSpecial New)

emjin committed:
Made it possible to run tests for all languages

Changed any_gen_of_string to Parse_generic.parse_pattern, also added debugging string to give more information when assert_bool fails

emjin committed:
Added java tests for constructors and id qualified

emjin committed:
Implemented inference for ArgKwd

When function arguments are labeled, display them appropriately. Also added a test

emjin committed:
Explicitly listed out languages for casing on booleans

Made this change so that if someone adds a language, they need to check what gets printed
